### ISSUE TITLE
Could not load OpenAI model. If you intended to use OpenAI, please check your OPENAI_API_KEY in spite of local

### ISSUE THREAD

**Issue by ErfolgreichCharismatisch** [cpacker/MemGPT#458]
**Describe the bug**
I set it all for local but when trying to run memgpt load directory --name BearbedBirb --input-dir "E:\Birbs" --recursive I got

```
Could not load OpenAI model. If you intended to use OpenAI, please check your OPENAI_API_KEY.
Original error:
No API key found for OpenAI.
Please set either the OPENAI_API_KEY environment variable or openai.api_key prior to initialization.
API keys can be found or created at https://platform.openai.com/account/api-keys

To disable the LLM entirely, set llm=None.
```

**Your setup (please complete the following information)**

- Your OS (Linux, MacOS, Windows): Windows 10
- Where you're trying to run MemGPT from
  - Terminal? PowerShell? cmd.exe? cmd
- Your python version (run `python --version`) 3.10
-  If you installed with `pip install pymemgpt`: Installed with pip

**Additional context**
Full output:

loading data
done loading data
LLM is explicitly disabled. Using MockLLM.
┌─────────────────────────────── Traceback (most recent call last) ────────────────────────────────┐
│ C:\Logs\Anaconda\envs\p310\lib\site-packages\llama_index\llms\utils.py:20 in resolve_llm         │
│                                                                                                  │
│   17 │   if llm == "default":                                                                    │
│   18 │   │   # return default OpenAI model. If it fails, return LlamaCPP                         │
│   19 │   │   try:                                                                                │
│ > 20 │   │   │   llm = OpenAI()                                                                  │
│   21 │   │   except ValueError as e:                                                             │
│   22 │   │   │   raise ValueError(                                                               │
│   23 │   │   │   │   "******\n"                                                                  │
│                                                                                                  │
│ C:\Logs\Anaconda\envs\p310\lib\site-packages\llama_index\llms\openai.py:92 in __init__           │
│                                                                                                  │
│    89 │   ) -> None:                                                                             │
│    90 │   │   additional_kwargs = additional_kwargs or {}                                        │
│    91 │   │                                                                                      │
│ >  92 │   │   api_key, api_type, api_base, api_version = resolve_openai_credentials(             │
│    93 │   │   │   api_key=api_key,                                                               │
│    94 │   │   │   api_type=api_type,                                                             │
│    95 │   │   │   api_base=api_base,                                                             │
│                                                                                                  │
│ C:\Logs\Anaconda\envs\p310\lib\site-packages\llama_index\llms\openai_utils.py:331 in             │
│ resolve_openai_credentials                                                                       │
│                                                                                                  │
│   328 │   api_version = api_version or openai.api_version or DEFAULT_OPENAI_API_VERSION          │
│   329 │                                                                                          │
│   330 │   if not api_key and api_type not in ("azuread", "azure_ad"):                            │
│ > 331 │   │   raise ValueError(MISSING_API_KEY_ERROR_MESSAGE)                                    │
│   332 │                                                                                          │
│   333 │   return api_key, api_type, api_base, api_version                                        │
│   334                                                                                            │
└──────────────────────────────────────────────────────────────────────────────────────────────────┘
ValueError: No API key found for OpenAI.
Please set either the OPENAI_API_KEY environment variable or openai.api_key prior to initialization.
API keys can be found or created at https://platform.openai.com/account/api-keys

During handling of the above exception, another exception occurred:

┌─────────────────────────────── Traceback (most recent call last) ────────────────────────────────┐
│ C:\Logs\Anaconda\envs\p310\lib\site-packages\memgpt\cli\cli_load.py:107 in load_directory        │
│                                                                                                  │
│   104 │   print("loading data")                                                                  │
│   105 │   docs = reader.load_data()                                                              │
│   106 │   print("done loading data")                                                             │
│ > 107 │   store_docs(name, docs)                                                                 │
│   108                                                                                            │
│   109                                                                                            │
│   110 @app.command("webpage")                                                                    │
│                                                                                                  │
│ C:\Logs\Anaconda\envs\p310\lib\site-packages\memgpt\cli\cli_load.py:30 in store_docs             │
│                                                                                                  │
│    27                                                                                            │
│    28 def store_docs(name, docs, show_progress=True):                                            │
│    29 │   """Common function for embedding and storing documents"""                              │
│ >  30 │   storage = StorageConnector.get_storage_connector(name=name)                            │
│    31 │   config = MemGPTConfig.load()                                                           │
│    32 │   embed_model = embedding_model()                                                        │
│    33                                                                                            │
│                                                                                                  │
│ C:\Logs\Anaconda\envs\p310\lib\site-packages\memgpt\connectors\storage.py:44 in                  │
│ get_storage_connector                                                                            │
│                                                                                                  │
│    41 │   │   if storage_type == "local":                                                        │
│    42 │   │   │   from memgpt.connectors.local import LocalStorageConnector                      │
│    43 │   │   │                                                                                  │
│ >  44 │   │   │   return LocalStorageConnector(name=name, agent_config=agent_config)             │
│    45 │   │                                                                                      │
│    46 │   │   elif storage_type == "postgres":                                                   │
│    47 │   │   │   from memgpt.connectors.db import PostgresStorageConnector                      │
│                                                                                                  │
│ C:\Logs\Anaconda\envs\p310\lib\site-packages\memgpt\connectors\local.py:57 in __init__           │
│                                                                                                  │
│    54 │   │   if len(self.nodes):                                                                │
│    55 │   │   │   self.index = VectorStoreIndex(self.nodes)                                      │
│    56 │   │   else:                                                                              │
│ >  57 │   │   │   self.index = EmptyIndex()                                                      │
│    58 │                                                                                          │
│    59 │   def get_nodes(self) -> List[TextNode]:                                                 │
│    60 │   │   """Get llama index nodes"""                                                        │
│                                                                                                  │
│ C:\Logs\Anaconda\envs\p310\lib\site-packages\llama_index\indices\empty\base.py:39 in __init__    │
│                                                                                                  │
│   36 │   │   **kwargs: Any,                                                                      │
│   37 │   ) -> None:                                                                              │
│   38 │   │   """Initialize params."""                                                            │
│ > 39 │   │   super().__init__(                                                                   │
│   40 │   │   │   nodes=None,                                                                     │
│   41 │   │   │   index_struct=index_struct or EmptyIndexStruct(),                                │
│   42 │   │   │   service_context=service_context,                                                │
│                                                                                                  │
│ C:\Logs\Anaconda\envs\p310\lib\site-packages\llama_index\indices\base.py:61 in __init__          │
│                                                                                                  │
│    58 │   │   │   else:                                                                          │
│    59 │   │   │   │   raise ValueError("nodes must be a list of Node objects.")                  │
│    60 │   │                                                                                      │
│ >  61 │   │   self._service_context = service_context or ServiceContext.from_defaults()          │
│    62 │   │   self._storage_context = storage_context or StorageContext.from_defaults()          │
│    63 │   │   self._docstore = self._storage_context.docstore                                    │
│    64 │   │   self._show_progress = show_progress                                                │
│                                                                                                  │
│ C:\Logs\Anaconda\envs\p310\lib\site-packages\llama_index\indices\service_context.py:157 in       │
│ from_defaults                                                                                    │
│                                                                                                  │
│   154 │   │   │   if llm_predictor is not None:                                                  │
│   155 │   │   │   │   raise ValueError("Cannot specify both llm and llm_predictor")              │
│   156 │   │   │   llm = resolve_llm(llm)                                                         │
│ > 157 │   │   llm_predictor = llm_predictor or LLMPredictor(                                     │
│   158 │   │   │   llm=llm, pydantic_program_mode=pydantic_program_mode                           │
│   159 │   │   )                                                                                  │
│   160 │   │   if isinstance(llm_predictor, LLMPredictor):                                        │
│                                                                                                  │
│ C:\Logs\Anaconda\envs\p310\lib\site-packages\llama_index\llm_predictor\base.py:92 in __init__    │
│                                                                                                  │
│    89 │   │   pydantic_program_mode: PydanticProgramMode = PydanticProgramMode.DEFAULT,          │
│    90 │   ) -> None:                                                                             │
│    91 │   │   """Initialize params."""                                                           │
│ >  92 │   │   self._llm = resolve_llm(llm)                                                       │
│    93 │   │                                                                                      │
│    94 │   │   if callback_manager:                                                               │
│    95 │   │   │   self._llm.callback_manager = callback_manager                                  │
│                                                                                                  │
│ C:\Logs\Anaconda\envs\p310\lib\site-packages\llama_index\llms\utils.py:22 in resolve_llm         │
│                                                                                                  │
│   19 │   │   try:                                                                                │
│   20 │   │   │   llm = OpenAI()                                                                  │
│   21 │   │   except ValueError as e:                                                             │
│ > 22 │   │   │   raise ValueError(                                                               │
│   23 │   │   │   │   "******\n"                                                                  │
│   24 │   │   │   │   "Could not load OpenAI model. "                                             │
│   25 │   │   │   │   "If you intended to use OpenAI, please check your OPENAI_API_KEY.\n"        │
└──────────────────────────────────────────────────────────────────────────────────────────────────┘
ValueError: ******
Could not load OpenAI model. If you intended to use OpenAI, please check your OPENAI_API_KEY.
Original error:
No API key found for OpenAI.
Please set either the OPENAI_API_KEY environment variable or openai.api_key prior to initialization.
API keys can be found or created at https://platform.openai.com/account/api-keys

To disable the LLM entirely, set llm=None.

**Comment by sarahwooders** [comment]
Can you share whats in your `~/.memgpt/configure` file, but with all the keys (e.g. OpenAI key, Azure keys) removed?

**Comment by sarahwooders** [comment]
Ok sorry looks like there was a bug, should be fixed by #462.

**Comment by cpacker** [comment]
Fixed with merge of #462, please let us know if it's still not working for you

**Comment by ErfolgreichCharismatisch** [comment]
It is working now, thanks for the prompt response.
